### ISSUE TITLE
Unexpected keyword argument 'seed'

### ISSUE THREAD

**Issue by Elmiar0642** [lightdock/lightdock#44]
**unexpected keyword argument 'seed'**
While running lightdock3_setup.py, the process stops with following error:
`.../Bio/pairwise2.py:278: BiopythonDeprecationWarning: Bio.pairwise2 has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.PairwiseAligner as a replacement, and contact the Biopython developers if you still need the Bio.pairwise2 module.
  warnings.warn(
[lightdock3_setup] INFO: Ignoring OXT atoms
[lightdock3_setup] INFO: Ignoring Hydrogen atoms
[lightdock3_setup] INFO: Ignoring water
[lightdock3_setup] INFO: Reading structure from 2UUY_rec.pdb PDB file...
[lightdock3_setup] INFO: 1628 atoms, 223 residues read.
[lightdock3_setup] INFO: Ignoring OXT atoms
[lightdock3_setup] INFO: Ignoring Hydrogen atoms
[lightdock3_setup] INFO: Ignoring water
[lightdock3_setup] INFO: Reading structure from 2UUY_lig.pdb PDB file...
[lightdock3_setup] INFO: 415 atoms, 55 residues read.
[lightdock3_setup] INFO: Calculating reference points for receptor 2UUY_rec.pdb...
[lightdock3_setup] INFO: Reference points for receptor found, skipping
[lightdock3_setup] INFO: Done.
[lightdock3_setup] INFO: Calculating reference points for ligand 2UUY_lig.pdb...
[lightdock3_setup] INFO: Reference points for ligand found, skipping
[lightdock3_setup] INFO: Done.
[lightdock3_setup] INFO: Saving processed structure to PDB file...
[lightdock3_setup] INFO: Done.
[lightdock3_setup] INFO: Saving processed structure to PDB file...
[lightdock3_setup] INFO: Done.
[lightdock3_setup] INFO: Calculating ANM for receptor molecule...
[lightdock3_setup] INFO: 10 normal modes calculated
[lightdock3_setup] INFO: Calculating ANM for ligand molecule...
[lightdock3_setup] INFO: 10 normal modes calculated
[lightdock3_setup] INFO: Calculating starting positions...
[lightdock3_setup] INFO:   * Surface density: TotalSASA/50.00
[lightdock3_setup] INFO:   * Swarm radius: 10.00 Å
[lightdock3_setup] INFO:   * 180° flip of 50% of starting poses: False
[lightdock3_setup] INFO:   * Ligand Max Diameter: 39.05 Å
[lightdock3_setup] INFO:   * Surface distance: 9.76 Å
Traceback (most recent call last):
  File "/home/randd/.local/bin/lightdock3_setup.py", line 142, in <module>
    starting_points_files = calculate_starting_positions(
  File ".../lightdock/prep/simulation.py", line 191, in calculate_starting_positions
    starting_points_files = calculate_initial_poses(
  File ".../lightdock/prep/poses.py", line 492, in calculate_initial_poses
    swarm_centers, receptor_diameter, ligand_diameter = calculate_surface_points(
  File ".../lightdock/prep/starting_points.py", line 156, in calculate_surface_points
    surface_clusters = kmeans2(data=coords, k=num_points, minit="points", seed=seed, iter=100)
TypeError: kmeans2() got an unexpected keyword argument 'seed'`

Upon clearing that error by changing the variable `seed` name at line number 70 to `spseed` on file `".../lightdock/prep/starting_points.py"` , I got another error given below:

`Traceback (most recent call last):
  File ".../.local/bin/lightdock3_setup.py", line 142, in <module>
    starting_points_files = calculate_starting_positions(
  File ".../.local/lib/python3.8/site-packages/lightdock/prep/simulation.py", line 191, in calculate_starting_positions
    starting_points_files = calculate_initial_poses(
  File ".../lightdock/prep/poses.py", line 492, in calculate_initial_poses
    swarm_centers, receptor_diameter, ligand_diameter = calculate_surface_points(
TypeError: calculate_surface_points() got an unexpected keyword argument 'seed'`

I fixed it by commented the line **500** on the `".../lightdock/prep/poses.py"`

But stillI got the previous error, in file ".../lightdock/prep/starting_points.py", upon which I removed the amends i Made, and commented or removed the seed parameter which is being passed to `kmeans2()` function on wherever it is called ( lines : 156, 251 ), And now it generates all files perfectly, and completes  the setup.

**To Reproduce**
Hmm.. Nothing to reproduce, I have explained it above.

**Expected behavior**
I made it work as expected. Here is the expected behaviour:

`randd  …/ProteinSynth/Protprotdocki/test  ♥ 10:51  lightdock3_setup.py 2UUY_rec.pdb 2UUY_lig.pdb --noxt -noh --now -anm/home/randd/.local/lib/python3.8/site-packages/Bio/pairwise2.py:278: BiopythonDeprecationWarning: Bio.pairwise2 has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.PairwiseAligner as a replacement, and contact the Biopython developers if you still need the Bio.pairwise2 module.
  warnings.warn(
[lightdock3_setup] INFO: Ignoring OXT atoms
[lightdock3_setup] INFO: Ignoring Hydrogen atoms
[lightdock3_setup] INFO: Ignoring water
[lightdock3_setup] INFO: Reading structure from 2UUY_rec.pdb PDB file...
[lightdock3_setup] INFO: 1628 atoms, 223 residues read.
[lightdock3_setup] INFO: Ignoring OXT atoms
[lightdock3_setup] INFO: Ignoring Hydrogen atoms
[lightdock3_setup] INFO: Ignoring water
[lightdock3_setup] INFO: Reading structure from 2UUY_lig.pdb PDB file...
[lightdock3_setup] INFO: 415 atoms, 55 residues read.
[lightdock3_setup] INFO: Calculating reference points for receptor 2UUY_rec.pdb...
[lightdock3_setup] INFO: Reference points for receptor found, skipping
[lightdock3_setup] INFO: Done.
[lightdock3_setup] INFO: Calculating reference points for ligand 2UUY_lig.pdb...
[lightdock3_setup] INFO: Reference points for ligand found, skipping
[lightdock3_setup] INFO: Done.
[lightdock3_setup] INFO: Saving processed structure to PDB file...
[lightdock3_setup] INFO: Done.
[lightdock3_setup] INFO: Saving processed structure to PDB file...
[lightdock3_setup] INFO: Done.
[lightdock3_setup] INFO: Calculating ANM for receptor molecule...
[lightdock3_setup] INFO: 10 normal modes calculated
[lightdock3_setup] INFO: Calculating ANM for ligand molecule...
[lightdock3_setup] INFO: 10 normal modes calculated
[lightdock3_setup] INFO: Calculating starting positions...
[lightdock3_setup] INFO:   * Surface density: TotalSASA/50.00
[lightdock3_setup] INFO:   * Swarm radius: 10.00 Å
[lightdock3_setup] INFO:   * 180° flip of 50% of starting poses: False
[lightdock3_setup] INFO:   * Ligand Max Diameter: 39.05 Å
[lightdock3_setup] INFO:   * Surface distance: 9.76 Å
[lightdock3_setup] INFO: Swarms after incompatible filter: 911
[lightdock3_setup] INFO: Swarms after interior points filter: 911
[lightdock3_setup] INFO: Generated 182 positions files
[lightdock3_setup] INFO: Done.
[lightdock3_setup] INFO: Number of calculated swarms is 182
[lightdock3_setup] INFO: Preparing environment
[lightdock3_setup] INFO: Done.
[lightdock3_setup] INFO: LightDock setup OK

randd  …/ProteinSynth/Protprotdocki/test  ♥ 10:51  ls
2UUY_lig.pdb                 swarm_1    swarm_109  swarm_119  swarm_129  swarm_139  swarm_149  swarm_159  swarm_169  swarm_179  swarm_25  swarm_35  swarm_45  swarm_55  swarm_65  swarm_75  swarm_85  swarm_95
2UUY_rec.pdb                 swarm_10   swarm_11   swarm_12   swarm_13   swarm_14   swarm_15   swarm_16   swarm_17   swarm_18   swarm_26  swarm_36  swarm_46  swarm_56  swarm_66  swarm_76  swarm_86  swarm_96
init                         swarm_100  swarm_110  swarm_120  swarm_130  swarm_140  swarm_150  swarm_160  swarm_170  swarm_180  swarm_27  swarm_37  swarm_47  swarm_57  swarm_67  swarm_77  swarm_87  swarm_97
lightdock_2UUY_lig_mask.npy  swarm_101  swarm_111  swarm_121  swarm_131  swarm_141  swarm_151  swarm_161  swarm_171  swarm_181  swarm_28  swarm_38  swarm_48  swarm_58  swarm_68  swarm_78  swarm_88  swarm_98
lightdock_2UUY_lig.pdb       swarm_102  swarm_112  swarm_122  swarm_132  swarm_142  swarm_152  swarm_162  swarm_172  swarm_19   swarm_29  swarm_39  swarm_49  swarm_59  swarm_69  swarm_79  swarm_89  swarm_99
lightdock_2UUY_rec_mask.npy  swarm_103  swarm_113  swarm_123  swarm_133  swarm_143  swarm_153  swarm_163  swarm_173  swarm_2    swarm_3   swarm_4   swarm_5   swarm_6   swarm_7   swarm_8   swarm_9
lightdock_2UUY_rec.pdb       swarm_104  swarm_114  swarm_124  swarm_134  swarm_144  swarm_154  swarm_164  swarm_174  swarm_20   swarm_30  swarm_40  swarm_50  swarm_60  swarm_70  swarm_80  swarm_90
lightdock_lig.nm.npy         swarm_105  swarm_115  swarm_125  swarm_135  swarm_145  swarm_155  swarm_165  swarm_175  swarm_21   swarm_31  swarm_41  swarm_51  swarm_61  swarm_71  swarm_81  swarm_91
lightdock_rec.nm.npy         swarm_106  swarm_116  swarm_126  swarm_136  swarm_146  swarm_156  swarm_166  swarm_176  swarm_22   swarm_32  swarm_42  swarm_52  swarm_62  swarm_72  swarm_82  swarm_92
setup.json                   swarm_107  swarm_117  swarm_127  swarm_137  swarm_147  swarm_157  swarm_167  swarm_177  swarm_23   swarm_33  swarm_43  swarm_53  swarm_63  swarm_73  swarm_83  swarm_93
swarm_0                      swarm_108  swarm_118  swarm_128  swarm_138  swarm_148  swarm_158  swarm_168  swarm_178  swarm_24   swarm_34  swarm_44  swarm_54  swarm_64  swarm_74  swarm_84  swarm_94

randd  …/ProteinSynth/Protprotdocki/test  ♥ 10:51  lightdock3.py setup.json 100 -c 1 -l 0/home/randd/.local/lib/python3.8/site-packages/Bio/pairwise2.py:278: BiopythonDeprecationWarning: Bio.pairwise2 has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.PairwiseAligner as a replacement, and contact the Biopython developers if you still need the Bio.pairwise2 module.
  warnings.warn(
[lightdock] INFO: simulation parameters saved to lightdock.info
[lightdock3_setup] INFO: Ignoring OXT atoms
[lightdock3_setup] INFO: Ignoring Hydrogen atoms
[lightdock3_setup] INFO: Ignoring water
[lightdock3_setup] INFO: Reading structure from lightdock_2UUY_rec.pdb PDB file...
[lightdock3_setup] INFO: 1628 atoms, 223 residues read.
[lightdock3_setup] INFO: Ignoring OXT atoms
[lightdock3_setup] INFO: Ignoring Hydrogen atoms
[lightdock3_setup] INFO: Ignoring water
[lightdock3_setup] INFO: Reading structure from lightdock_2UUY_lig.pdb PDB file...
[lightdock3_setup] INFO: 415 atoms, 55 residues read.
[lightdock] INFO: Loading scoring function...
[lightdock] INFO: Using DFIRE scoring function
[lightdock] INFO: Done.
[kraken] INFO: Kraken has 1 tentacles (cpu cores)
[kraken] INFO: Tentacle ready with 1 tasks
[kraken] INFO: 1 ships ready to be smashed
[lightdock] INFO: Monster spotted
[kraken] INFO: Release the Kraken!
[0] step 1
[0] step 2
[0] step 3
[0] step 4
[0] step 5
[0] step 6
[0] step 7
[0] step 8
[0] step 9
[0] step 10
[0] step 11
[0] step 12
[0] step 13
[0] step 14
[0] step 15
[0] step 16
[0] step 17
[0] step 18
[0] step 19
[0] step 20
[0] step 21
[0] step 22
[0] step 23
[0] step 24
[0] step 25
[0] step 26
[0] step 27
[0] step 28
[0] step 29
[0] step 30
[0] step 31
[0] step 32
[0] step 33
[0] step 34
[0] step 35
[0] step 36
[0] step 37
[0] step 38
[0] step 39
[0] step 40
[0] step 41
[0] step 42
[0] step 43
[0] step 44
[0] step 45
[0] step 46
[0] step 47
[0] step 48
[0] step 49
[0] step 50
[0] step 51
[0] step 52
[0] step 53
[0] step 54
[0] step 55
[0] step 56
[0] step 57
[0] step 58
[0] step 59
[0] step 60
[0] step 61
[0] step 62
[0] step 63
[0] step 64
[0] step 65
[0] step 66
[0] step 67
[0] step 68
[0] step 69
[0] step 70
[0] step 71
[0] step 72
[0] step 73
[0] step 74
[0] step 75
[0] step 76
[0] step 77
[0] step 78
[0] step 79
[0] step 80
[0] step 81
[0] step 82
[0] step 83
[0] step 84
[0] step 85
[0] step 86
[0] step 87
[0] step 88
[0] step 89
[0] step 90
[0] step 91
[0] step 92
[0] step 93
[0] step 94
[0] step 95
[0] step 96
[0] step 97
[0] step 98
[0] step 99
[0] step 100
[kraken] INFO: folding tentacle Tentacle-1
[kraken] INFO: 1 ships destroyed
[lightdock] INFO: Finished.`

**Screenshots**
I don't have time to add screenshot now. just came to inform the community about this.

**Desktop :**
 - OS: Ubuntu 
 - Version 20.04.1

**Additional context**
Nothing

**Comment by brianjimenez** [comment]
Thanks @Elmiar0642 for reporting this issue, it seems the `kmeans2()` function has a different signature in your library version. Please, could you provide a `pip freeze` output or similar to see the installed Python library versions? Also Python interpreter version would be necessary.

**Comment by Elmiar0642** [comment]
Sure @brianjimenez , here is the output:
```

absl-py==1.3.0
adal==1.2.7
addict==2.4.0
aiohttp==3.8.3
aiosignal==1.3.1
alabaster==0.7.8
alembic==1.8.1
anyio==3.6.2
applicationinsights==0.11.9
apptools==4.5.0
apturl==0.5.2
argcomplete==1.12.3
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
asttokens==2.0.8
astunparse==1.6.3
async-timeout==4.0.2
attrdict==2.0.1
attrs==22.1.0
autobahn==17.10.1
Automat==0.8.0
autopep8==2.0.0
azure-cli-core==2.34.1
azure-cli-telemetry==1.0.6
azure-core==1.26.1
azure-mgmt-core==1.3.2
Babel==2.6.0
backcall==0.2.0
bcrypt==3.1.7
beautifulsoup4==4.11.1
biopython==1.80
bleach==5.0.1
blinker==1.4
blis==0.7.9
Boruta==0.3
boto3==1.26.45
botocore==1.29.45
Brlapi==0.7.0
cachetools==5.2.0
catalogue==1.0.2
cbor==1.0.0
certifi==2019.11.28
cffi==1.15.1
chardet==3.0.4
charset-normalizer==2.1.1
ci-info==0.3.0
Click==7.0
cloudpickle==2.2.0
colorama==0.4.3
colorlover==0.3.0
command-not-found==0.3
commentjson==0.9.0
configobj==5.0.6
constantly==15.1.0
contourpy==1.0.5
cryptography==38.0.3
cufflinks==0.17.3
cupshelpers==1.0
cupy-cuda101==9.6.0
cycler==0.11.0
cymem==2.0.7
Cython==0.29.14
databricks-cli==0.17.3
dataframe-image==0.1.3
datasets==2.8.0
dblatex===0.3.11py3
dbus-python==1.2.16
debugpy==1.6.3
decorator==5.1.1
defer==1.0.6
defusedxml==0.7.1
deprecation==2.1.0
dill==0.3.6
distro==1.8.0
distro-info===0.23ubuntu1
docker==5.0.3
docker-compose==1.29.2
dockerpty==0.4.1
docopt==0.6.2
docutils==0.16
duplicity==0.8.12.0
entrypoints==0.3
envisage==4.9.0
et-xmlfile==1.1.0
etelemetry==0.3.0
executing==1.1.1
fasteners==0.14.1
fastjsonschema==2.16.2
fastrlock==0.8
filelock==3.8.0
Flask==2.2.2
flatbuffers==22.9.24
fonttools==4.38.0
freesasa==2.1.0
frozenlist==1.3.3
fsleyes==1.5.0
fsleyes-props==1.8.2
fsleyes-widgets==0.12.3
fslpy==3.10.0
fsspec==2022.11.0
fstrings==0.1.0
funcy==1.17
fuse-python==1.0.0
future==0.18.2
gast==0.4.0
gensim==3.8.3
gitdb==4.0.9
GitPython==3.1.29
google-auth==2.13.0
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
greenlet==2.0.0
grpcio==1.50.0
gunicorn==20.1.0
gym==0.26.2
gym-notices==0.0.8
h5py==3.7.0
html5lib==1.0.1
htmlmin==0.1.12
httplib2==0.14.0
huggingface-hub==0.11.1
humanfriendly==10.0
hyperlink==19.0.0
idna==2.8
ImageHash==4.3.1
imageio==2.22.4
imagesize==1.2.0
imbalanced-learn==0.7.0
importlib-metadata==5.0.0
importlib-resources==5.10.0
incremental==16.10.1
iotedgedev==3.3.6
iotedgehubdev==0.14.18
ipykernel==6.16.2
ipynb-py-convert==0.4.6
ipython==8.5.0
ipython-genutils==0.2.0
ipywidgets==8.0.2
isodate==0.6.1
itsdangerous==2.1.2
jedi==0.18.1
Jinja2==3.1.2
jmespath==1.0.1
joblib==1.2.0
json5==0.9.10
jsonpath-rw==1.4.0
jsonschema==3.2.0
jupyter-client==7.4.4
jupyter-console==6.0.0
jupyter-core==4.11.2
jupyter-packaging==0.12.3
jupyter-server==1.21.0
jupyterlab==3.5.0
jupyterlab-pygments==0.2.2
jupyterlab-server==2.16.3
jupyterlab-widgets==3.0.3
keras==2.10.0
Keras-Preprocessing==1.1.2
keyring==18.0.1
kiwisolver==1.4.4
kmodes==0.12.2
knack==0.9.0
language-selector==0.1
lark-parser==0.7.8
laspy==2.3.0
launchpadlib==1.10.13
lazr.restfulclient==0.14.2
lazr.uri==1.0.3
libclang==14.0.6
lightdock==0.9.3.post1
lightgbm==3.3.3
llvmlite==0.39.1
lockfile==0.12.2
looseversion==1.0.2
louis==3.12.0
lxml==4.9.1
lz4==3.0.2+dfsg
macaroonbakery==1.3.1
Mako==1.1.0
Markdown==3.4.1
MarkupSafe==2.1.1
matplotlib==3.5.3
matplotlib-inline==0.1.6
mayavi==4.7.1
meson==0.53.2
missingno==0.5.1
mistune==2.0.4
mlflow==1.30.0
mlxtend==0.21.0
monotonic==1.5
more-itertools==8.0.2
mpi4py==3.0.3
msal==1.20.0
msal-extensions==0.3.1
msrest==0.7.1
msrestazure==0.6.4
multidict==6.0.2
multimethod==1.9
multiprocess==0.70.14
murmurhash==1.0.9
nbclassic==0.4.5
nbclient==0.7.0
nbconvert==7.2.2
nbformat==5.7.0
nest-asyncio==1.5.6
netifaces==0.10.4
networkx==2.8.7
nibabel==4.0.2
nipype==1.8.5
nltk==3.7
notebook==6.5.1
notebook-shim==0.2.0
numba==0.56.4
numexpr==2.8.4
numpy==1.23.4
oauthlib==3.1.0
olefile==0.46
open3d==0.13.0
opencv-contrib-python==4.6.0.66
openpyxl==3.0.10
openshot-qt==2.4.3
opt-einsum==3.3.0
packaging==21.3
pandas==1.5.1
pandas-profiling==3.4.0
pandastable==0.13.0
pandocfilters==1.5.0
paramiko==2.6.0
parso==0.8.3
patsy==0.5.3
pexpect==4.6.0
phik==0.12.2
pickleshare==0.7.5
Pillow==9.3.0
pkginfo==1.8.3
pkgutil-resolve-name==1.3.10
plac==1.1.3
plotly==5.11.0
ply==3.11
plyfile==0.7.4
portalocker==1.7.1
preshed==3.0.8
ProDy==2.0.1
progressbar==2.5
prometheus-client==0.15.0
prometheus-flask-exporter==0.20.3
prompt-toolkit==3.0.31
protobuf==3.19.6
prov==2.0.0
psutil==5.9.3
ptyprocess==0.7.0
pure-eval==0.2.2
py-ubjson==0.14.0
pyarrow==10.0.1
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycairo==1.16.2
pycaret==2.3.10
pycodestyle==2.9.1
pycparser==2.21
pycups==1.9.73
pydantic==1.10.2
pydicom==1.4.1
pydot==1.4.2
pyface==6.1.2
Pygments==2.13.0
PyGObject==3.36.0
PyHamcrest==1.9.0
PyJWT==2.6.0
pyLDAvis==3.3.1
pymacaroons==0.13.0
pymol==2.3.0
PyNaCl==1.3.0
pynndescent==0.5.8
pyod==1.0.6
PyOpenGL==3.1.6
pyOpenSSL==22.0.0
pyparsing==2.4.7
pypng==0.0.20
PyQRCode==1.2.1
PyQt5==5.14.1
pyqtgraph==0.13.1
pyRFC3339==1.1
pyrsistent==0.18.1
pysurfer==0.9.0
python-apt==2.0.0+ubuntu0.20.4.8
python-dateutil==2.8.2
python-debian===0.1.36ubuntu1
python-dotenv==0.21.0
python-snappy==0.5.3
PyTrie==0.2
pytz==2022.5
PyWavelets==1.4.1
pyxdg==0.26
PyYAML==5.4.1
pyzmq==24.0.1
querystring-parser==1.2.4
ranger-fm==1.9.3
rdflib==6.2.0
rdkit==2022.9.1
regex==2022.10.31
reportlab==3.5.34
requests==2.28.1
requests-oauthlib==1.3.1
requests-unixsocket==0.2.0
responses==0.18.0
roman==2.0.0
rsa==4.9
s3transfer==0.6.0
sacremoses==0.0.53
scikit-image==0.19.3
scikit-learn==1.1.3
scikit-plot==0.3.7
scipy==1.5.4
screen-resolution-extra==0.0.0
seaborn==0.12.1
SecretStorage==2.3.1
Send2Trash==1.8.0
sentencepiece==0.1.97
service-identity==18.1.0
shap==0.41.0
simplejson==3.16.0
sip==4.19.21
six==1.14.0
sklearn==0.0
slicer==0.0.7
smart-open==6.2.0
smmap==5.0.0
sniffio==1.3.0
soupsieve==2.3.2.post1
spacy==2.3.8
Sphinx==1.8.5
SQLAlchemy==1.4.42
sqlparse==0.4.3
srsly==1.0.6
stack-data==0.5.1
statsmodels==0.13.4
systemd-python==234
tabulate==0.9.0
tangled-up-in-unicode==0.2.0
tenacity==8.1.0
tensorboard==2.10.1
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tensorflow-estimator==2.10.0
tensorflow-gpu==2.10.0
tensorflow-io-gcs-filesystem==0.27.0
termcolor==2.0.1
terminado==0.17.0
testpath==0.4.4
textblob==0.17.1
texttable==1.6.4
Theano==1.0.5
thinc==7.4.6
threadpoolctl==3.1.0
tifffile==2022.10.10
tinycss2==1.2.1
tk==0.1.0
tokenizers==0.13.2
tomli==2.0.1
tomlkit==0.11.6
torch==1.12.1
torch-cluster==1.6.0
torch-geometric==2.1.0.post1
torch-scatter==2.0.9
torch-sparse==0.6.15
torch-spline-conv==1.2.1
torchaudio==0.12.1
torchvision==0.13.1
tornado==6.2
tqdm==4.64.1
traitlets==5.5.0
traits==6.3.2
traitsui==6.1.3
transformers==4.25.1
turtle==0.0.1
Twisted==18.9.0
txaio==2.10.0
typing-extensions==4.4.0
u-msgpack-python==2.1
ubuntu-advantage-tools==27.12
ubuntu-drivers-common==0.0.0
ufw==0.36
umap-learn==0.5.3
unattended-upgrades==0.1
urllib3==1.26.12
usb-creator==0.3.7
visions==0.7.5
wadllib==1.3.3
wasabi==0.10.1
wcwidth==0.2.5
webencodings==0.5.1
websocket-client==0.59.0
Werkzeug==2.2.2
widgetsnbextension==4.0.3
wordcloud==1.8.2.2
wrapt==1.14.1
wsaccel==0.6.2
wxPython==4.2.0
xgboost==1.6.2
xkit==0.0.0
xlrd==2.0.1
xxhash==3.2.0
yarl==1.8.1
yellowbrick==1.5
zipp==3.10.0
zope.interface==4.7.1
```

**Comment by alchemistcai** [comment]
scipy's version should >= [1.7.0](https://docs.scipy.org/doc/scipy/release.1.7.0.html#scipy-cluster-improvements) to make kmeans2 supporting the `seed` keyword.

pip's requirement is >=1.3.1,however.

![Screenshot_20230117_142342](https://user-images.githubusercontent.com/60164864/212825220-43d2b09e-628e-4f7a-b573-1c6ab542729f.png)

**Comment by brianjimenez** [comment]
What about the Python version? In my local environment I'm running scipy version `1.8.1` while the last [stable](https://pypi.org/project/scipy/#history) seems to be `1.10.1`.

`scipy.cluster.vq.kmeans2()` has a different [signature](https://docs.scipy.org/doc/scipy-1.5.4/reference/generated/scipy.cluster.vq.kmeans2.html#scipy.cluster.vq.kmeans2) in version `1.5.4`, this requires fixing scipy library version in setup.py to at least `1.7.0`.

**Comment by JorgeRoel** [comment]
You may want to make a fresh lightdock installation inside a dedicated python environment to avoid conflicts with other python libraries within your system:

`python3 -m venv /path/to/new/virtual/environment
source <venv>/bin/activate
pip install lightdock`

**Comment by brianjimenez** [comment]
Issue addressed at [0.9.3.post2 release](https://github.com/lightdock/lightdock/releases/tag/0.9.3.post2).
